File: FindClosestNumbertoZero.java
class Solution {
    public int findClosestNumber(int[] nums) {
        int val = nums[0];

        for (int num: nums){
            if (Math.abs(num) < Math.abs(val)){
                val = num;
            }
            else if(Math.abs(num) == Math.abs(val) && num > val){
                val = num;
            }
        }
        return val;
    }
    // Time: O(n)
    // Space: O(1)
}

File: FindClosestNumbertoZero.py
class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        val = nums[0]

        for num in nums:
            if abs(num) < abs(val):
                val = num
            elif abs(num) == abs(val) and num > val:
                val = num
    
        return val

        # Time: O(n)
        # Space: O(1)
